File: hard/StringList.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

class StringList
{
    static SortedSet<string> words = new SortedSet<string>();
    static string letters;
    static int wordSize;

    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        {
            while (!reader.EndOfStream)
            {
                string[] parts = reader.ReadLine().Split(',');
                wordSize = Int32.Parse(parts[0]);
                letters = parts[1];
                words.Clear();

                BuildWords("");

                Console.WriteLine(string.Join(",", words));
            }
        }
    }

    static void BuildWords(string word)
    {
        if (word.Length < wordSize)
            foreach (char c in letters)
                BuildWords(word + c);
        else
            words.Add(word);
    }
}

File: hard/StringList.java

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.Iterator;
import java.util.Set;
import java.util.TreeSet;

public class Main {

    static class StringList implements Iterable<String> {

        Set<String> words;
        char[] letters;
        int wordSize;

        public StringList(String l, int ws) {
            letters = l.toCharArray();
            wordSize = ws;
            words = new TreeSet<>();
            BuildWords("");
        }

        private void BuildWords(String word) {
            if (word.length() < wordSize) {
                for (char c : letters) {
                    BuildWords(word + c);
                }
            } else {
                words.add(word);
            }
        }

        @Override
        public Iterator<String> iterator() {
            return words.iterator();
        }
        
    }

    public static void main(String[] args) throws FileNotFoundException, IOException {
        BufferedReader r = new BufferedReader(new FileReader(args[0]));
        String line;
        while ((line = r.readLine()) != null) {
            String[] parts = line.split(",");

            StringList list = new StringList(parts[1], Integer.parseInt(parts[0]));

            Boolean comma = false;
            for (String word : list) {
                if (comma) {
                    System.out.print(",");
                }
                comma = true;

                System.out.print(word);
            }
            System.out.println();
        }
        r.close();
    }
}
